File: app/src/main/java/com/example/parksmartapipull/MainActivity.java
package com.example.parksmartapipull;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.android.volley.Response;
import com.android.volley.VolleyError;

import org.json.JSONObject;

// for more reference, see https://developer.android.com/training/volley/simple
// I have included examples for both java and kotlin
public final class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final TextView outputView = findViewById(R.id.test);

        ParkSmartServer server = new ParkSmartServer(this);
        server.pull("Lot_D", new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {
                String output = "(java) Response is: " + response.toString();
                outputView.setText(output);
            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                String output = "(java) HTTP request encountered Error: " + error.getMessage();
                outputView.setText(output);
            }
        });
    }
}

File: app/src/main/java/com/example/parksmartapipull/MainActivity.kt
package com.example.parksmartapipull

import android.app.DownloadManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.android.volley.toolbox.Volley
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.VolleyError
import com.android.volley.toolbox.JsonArrayRequest
import com.android.volley.toolbox.JsonObjectRequest
import org.json.JSONArray
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // get handle for the output view where the response should be displayed
        val outputView = findViewById<TextView>(R.id.test)

        //form request using API call
        var server = ParkSmartServer(this)
        server.pull("Lot_D", Response.Listener { response ->
            //display the response from the request
            outputView.text = "(kotlin) Response is: ${response.toString()}"
        }, Response.ErrorListener { error: VolleyError ->
            outputView.text = "(kotlin) HTTP request encountered Error: ${error.message}"
        })
    }

}
